### ISSUE TITLE
C# support not working after C# extension installed

### ISSUE THREAD

**Issue by leocrespo** [Microsoft/vscode#4467]
- VSCode Version: 0.10.11
- OS Version: 10.11.3

Steps to Reproduce:

1. Download this file and open it with VS Code http://d.pr/f/1jczl
2. Check if there's intellisense.

If it doesn't have intellisense, it's not an editor. Might as well use textedit.

![Image of No Intellisense]
(http://d.pr/i/1he2V.gif)

**Comment by Tyriar** [comment]
Did you install the C# extension? https://code.visualstudio.com/Docs/languages/csharp#_installing-c35-support

**Comment by leocrespo** [comment]
Yes it's installed

![Image of VSCode]
(http://d.pr/i/15tLS.png)

**Comment by Tyriar** [comment]
@leocrespo and was vscode restarted after it was installed?

**Comment by leocrespo** [comment]
Yes of course. Have you tried downloading the file and opening it, I'm sure there won't be any intellisense.

**Comment by Tyriar** [comment]
Works fine for me. I'm on Windows 10 and had VS2015 installed before VSCode.

![image](https://cloud.githubusercontent.com/assets/2193314/13902545/8c9a1062-ee09-11e5-83a0-e56936a1c6f7.png)

I haven't actually used the C# support before, one of the other devs might be able to help you out beyond this point.

**Comment by leocrespo** [comment]
I can do that too, that's not intellisense. 
![Image of candothattoo]
(http://d.pr/i/1k46V.png)

This is intellisense
![Image of thisisintellisense]
(http://d.pr/i/XTlX.gif)

**Comment by Tyriar** [comment]
@leocrespo this might resolve your issue http://stackoverflow.com/a/29979730/1156119

**Comment by leocrespo** [comment]
1. There isn't any option for me to create a project.json or a solution. (amazing)
2. There is no "Open folder" command 

![Image of selection]
(http://d.pr/i/aeyZ.gif)

**Comment by leocrespo** [comment]
All I want is to have basic intellisense on the file I posted, just as I do when I open it with VS. So I created a solution in VS2015. 

![Image of aa]
(http://d.pr/i/14zVQ.png)

then opened everything with VSCode and still no intellisense. 
![Image of bb]
(http://d.pr/i/14zVQ.gif)

**Comment by Tyriar** [comment]
Visual Studio can generate sln files as you mention, VSCode cannot. VSCode typically relies on external tools or extensions for that kind of project scaffolding.

As for the `project.json`, it looks like this is created using the relatively new .NET Core and [`dotnet` CLI](http://dotnet.github.io/getting-started/) which uses a more folder-based approach to C# development which is more inline with what VSCode was designed to work with (as opposed to project/solution-based).

So once the .NET Core is installed, you should be able to run this to generate a project.json:

```
dotnet new
```

And then this to run your application:

```
dotnet restore # install deps
dotnet run
```

When coming back to VSCode I ran into this OmniSharp issue related to VSCode not being able to find the globally installed dnx runtime https://github.com/Microsoft/vscode/issues/126#issuecomment-160131771

---

Not sure why open folder is not there, it always has been for me (I don't use OSX though).

![image](https://cloud.githubusercontent.com/assets/2193314/13903312/6a670562-ee32-11e5-9244-e22b8301c335.png)

You can always run the command from the command palette:

1. Press <kbd>F1</kbd>
2. Type `open folder`
3. Press <kbd>enter</kbd>

![image](https://cloud.githubusercontent.com/assets/2193314/13903317/8e364188-ee32-11e5-8c6a-e4d929e5df56.png)

**Comment by pflannery** [comment]
@leocrespo just want to point out that the example you show here would never work as your trying to write the script in the c# class root scope.
In this scenario Visual studio also fails to show intellisense too

![animation](https://cloud.githubusercontent.com/assets/1727302/13905713/94439ea8-eebf-11e5-8228-9d51b603061d.gif)


Try this example project in vscode https://github.com/dotnet/core/tree/master/samples/helloworld

run `dotnet restore` in the root and you should get this:

![image](https://cloud.githubusercontent.com/assets/1727302/13905764/e5085166-eec0-11e5-87b0-8fa97477751b.png)

i'm running vscode 0.10.11

**Comment by jrieken** [comment]
@leocrespo There is a Omnisharp log file (Cmd+L,L) Please paste its contents here.

**Comment by jrieken** [comment]
This issue was moved to OmniSharp/omnisharp-vscode#114

**Comment by leocrespo** [comment]
There's no need to send logs or try any other steps, this problem takes 15 seconds to reproduce. 
Just download this file, open it with VS Code and you'll see there's no intellisense 

http://d.pr/f/1jczl

**Comment by jrieken** [comment]
as outlined by @pflannery [here](https://github.com/Microsoft/vscode/issues/4467#issuecomment-198968898), what you trying to do isn't valid C# so you should not expect IntelliSense

**Comment by leocrespo** [comment]
VSCode, no intellisense
![Image of aaa](http://d.pr/i/FgnW.gif)

VS 2015, same file, here's the intellisense. 
![Image of aaa](http://d.pr/i/1emfN.gif)

**Comment by yankun007** [comment]
I can't Install C#...with ext install C# . nothing happened.

**Comment by Tyriar** [comment]
@yankun007 I believe it's quite large so it sometimes takes 30-60 seconds to download. Are you sure you waited long enough?

**Comment by dave-hillier** [comment]
I had this problem. I took the following steps and it started working again.

1. uninstall all the plugins
2. uninstall VSCode
3. reinstall and open the project in VS Code
4. reinstall the c#
5. still not working after reload
6. Restart & reopen
it started downloading Omnisharp and added some files to my project. 

I think its unfair that this was closed, but it looks like some kind of setup problem.
